overlord/servicestate/quota_control.go

```diff
@@ -220,6 +238,13 @@ func UpdateQuota(st *state.State, name string, updateOpts QuotaGroupUpdate) erro
 	st.Lock()
 	defer st.Unlock()
 
+	// TODO: remove again once UpdateQuota is no longer exported
+	//       If creation is blocked manipulation and removal do not
+	//       need extra checks.
+	if err := quotaGroupsAvailable(st); err != nil {
```

Comment:
I think we should add feature check also to RemoveQuota() in case the feature gets disabled after having it ON.

Reply:
Happy to do this - was thinking maybe blocking create is enough because then there is nothing to manipulate or remove. But then you can turn this on/off so it quickly becomes messy. So probably best to add it to remove too.

Reply:
I think the conclusion after the standup is to leave it as is, e.g. allow Remove.

Reply:
We could even remove this one but no strong opinion here.

Reply:
it's not really used in API at moment so a bit immaterial? but I agree that maybe it should be left out